### PR TITLE
Add basic command line - build-artifacts option

### LINKED ISSUE #4: Add command line tool
The tool should start with this sub command:

- oci build-artifacts CHANGE
  Start build-artifacts using jenknins.run() and watch the job until it completes.

Gloabl options:
- --verbose - enable debug log

### PR DESCRIPTION
**Pull request by bell-lev**
Sub command:
    build-artifacts {change-number}

Global options:
    --verbose - enable debug log

usage example:
`oci build-artifacts 98719`

resolves #4

### CODE REVIEW

File: oci.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import argparse
+import logging
+
+from oci import config
+from oci import gerrit
+from oci import jenkins
```
Comment (nirs) [inline oci.py]: I think we should move this to oci/main.py.

The way to run the tool is to have a script called "oci", that will do:

```
#!/usr/bin/python
from oci import main
main.run()
```

This file will be installed in the path to make the tool available from the shell, while the oci
package is installed in python site-packages directory.

File: oci.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+
+def main():
```
Comment (nirs) [inline oci.py]: If this file becomes ocu/main.py, a good name for this is run().

File: oci.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        help='number of the patch to run')
+
+    args = parser.parse_args()
+    args.command(args)
+
+
+def run_build_artifacts(args):
+    logging.basicConfig(
+        level=logging.INFO,
+        format="%(asctime)s %(levelname)-7s [%(name)s] %(message)s")
```
Comment (nirs) [inline oci.py]: Configuring the logger should be done in main(), not in the sub command. A sub command should only care about
the actual command. Everything else should be handled by main.

File: oci.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    job_url = ja.wait_for_queue(queue_url)
+
+    log.info("[ 4/5 ] Waiting for job %s", job_url)
+    result = ja.wait_for_job(job_url)
+
+    log.info("[ 5/5 ] Job completed with %s", result)
+
+
+if __name__ == "__main__":
+    main()
```
Comment (nirs) [inline oci.py]: This is not needed in oci/main.py, or oci script.

File: oci.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+
+def run_build_artifacts(args):
+    logging.basicConfig(
+        level=logging.INFO,
+        format="%(asctime)s %(levelname)-7s [%(name)s] %(message)s")
+
+    log = logging.getLogger("build-artifacts")
+    if not args.verbose:
+        logging.disable(logging.CRITICAL)
```
Comment (nirs) [inline oci.py]: --verbose usually mean:

    logging.basicConfig(
        level=logging.DEBUG if args.verbose else logging.INFO,
        ...

File: oci.py [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    args = parser.parse_args()
+    args.command(args)
+
+
+def run_build_artifacts(args):
+    logging.basicConfig(
+        level=logging.INFO,
+        format="%(asctime)s %(levelname)-7s [%(name)s] %(message)s")
+
+    log = logging.getLogger("build-artifacts")
```
Comment (nirs) [inline oci.py]: I think we need single logger in the module level, called "oci".

File: oci.py [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    parser.add_argument(
+        '--verbose',
+        help="increase output verbosity",
+        action="store_true")
+    subparsers = parser.add_subparsers(title="commands")
+
+    build_artifacts = subparsers.add_parser(
+        "build-artifacts",
+        help="start the build-artifacts stage")
+    build_artifacts.set_defaults(command=run_build_artifacts)
```
Comment (nirs) [inline oci.py]: This will be nicer if the function name is build_artifacts.

File: oci.py [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+    parser.add_argument(
+        '--verbose',
+        help="increase output verbosity",
+        action="store_true")
+    subparsers = parser.add_subparsers(title="commands")
+
+    build_artifacts = subparsers.add_parser(
+        "build-artifacts",
+        help="start the build-artifacts stage")
```
Comment (nirs) [inline oci.py]: This actually start build artifacts and wait until it completes. But I think the help can be something like "build artifacts for a change"

**Review by nirs — CHANGES_REQUESTED** [review]
Thanks Bell

**Comment by bell-lev** [comment]
@nirs I named the script oci_script for now. If I understood correctly the oci script should be outside of the oci folder? I am afraid the names clash and the folder should be renamed?

File: oci.py (hunk 1 above)
Comment (bell-lev) [reply oci.py]: Done, but please see my comment on this.

File: oci.py (hunk 2 above)
Comment (bell-lev) [reply oci.py]: Done.

File: oci.py (hunk 3 above)
Comment (bell-lev) [reply oci.py]: I removed the logger, please let me know if you have different intentions.

File: oci.py (hunk 4 above)
Comment (bell-lev) [reply oci.py]: Done.

File: oci.py (hunk 7 above)
Comment (bell-lev) [reply oci.py]: Done.

File: oci/main.py [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+    subparsers = parser.add_subparsers(title="commands")
+
+    build_artifacts_parser = subparsers.add_parser(
+        "build-artifacts",
+        help="build artifacts for a change")
+    build_artifacts_parser.set_defaults(command=build_artifacts)
+
+    build_artifacts_parser.add_argument(
+        'change',
+        help='number of the patch to run')
```
Comment (nirs) [inline oci/main.py]: This sounds like the number of patches we are running, while we build artifacts for a single patch.

How about "Gerrit change number?

File: oci/main.py [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    out.info(("queue", queue_url))
+    job_url = ja.wait_for_queue(queue_url)
+
+    out.step("Waiting until job is completed")
+    out.info(("job", job_url))
+    result = ja.wait_for_job(job_url)
+
+    if result != "SUCCESS":
+        out.failure("Build artifcats failed with %s", result)
+        sys.exit(1)
```
Comment (nirs) [inline oci/main.py:67]: You forgot the last line:

    out.success("Job completed successfully, congratulations!")

File: oci_script [hunk 11]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/python
+from oci import main
+main.run()
```
Comment (nirs) [inline oci_script]: Please add a new line to the end of the file.

The file should be named "oci", and be executable.

You should be able to run it from source like this:

    ./oci build-artifacts 12345

**Review by nirs — CHANGES_REQUESTED** [review]

File: oci_script (hunk 11 above)
Comment (nirs) [reply oci_script]: Lets rename the script to ovirt-ci:

    ovirt-ci build-artifacts 12345

See also #31

File: oci/main.py (hunk 9 above)
Comment (bell-lev) [reply oci/main.py]: Done

File: oci/main.py (hunk 10 above)
Comment (bell-lev) [reply oci/main.py:67]: Done

File: oci_script (hunk 11 above)
Comment (bell-lev) [reply oci_script]: Done

**Review by nirs — APPROVED** [review]
Thanks Bell!

**Pull request merged**